File: 13. Rekursi Lanjut/D. Brankas Terkunci/d_brankas_terkunci.pas
program d_brankas_terkunci;

var
N, K, i : integer;
isVisited: array[1..17] of boolean;
save: array[1..17] of integer;

procedure draw(depth : integer);
var
j, l : integer;
begin
	if (depth >= K) then begin
		for j := 1 to K do write(save[j], ' ');
		writeln;
	end
	else begin
		for l := (save[depth]+1) to N do begin
			if (not isVisited[l]) then begin
				isVisited[l] := true;
				save[depth+1] := l;
				draw(depth + 1);
				isVisited[l] := false;
			end;
		end;
	end;
end;

begin
	readln(N, K);
	for i := 1 to 17 do isVisited[i] := false;

	for i:= 1 to N do begin
		save[1] := i;
		draw(1);
	end;
end.

File: 13. Rekursi Lanjut/D. Brankas Terkunci/d_brankas_terkunci.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int N, K;
bool isVisited[17];
int save[17];

void draw(int depth)
{
    if (depth >= K - 1)
    {
        int i;
        for (i = 0; i < K; i++)
        {
            printf("%d ", save[i] + 1);
        }
        printf("\n");
    }
    else
    {
        int i;
        for (i = save[depth] + 1; i < N; i++)
        {
            if (!isVisited[i])
            {
                isVisited[i] = true;
                save[depth + 1] = i;
                draw(depth + 1);
                isVisited[i] = false;
            }
        }
    }
}

int main()
{
    scanf("%d %d", &N, &K);
    memset(isVisited, false, sizeof(isVisited));

    int i;
    for (i = 0; i < N; i++)
    {
        save[0] = i;
        draw(0);
    }
    return 0;
}
